src/rust/generators/sysmon-generator/src/serialization.rs

```diff
@@ -26,70 +26,66 @@ impl CheckedError for SysmonDecoderError {
             Self::DeserializeError(_) => Recoverable::Persistent,
             Self::DecompressionError(_) => Recoverable::Persistent,
             Self::TimeError(_) => Recoverable::Persistent,
+            Self::Utf8Error(_) => Recoverable::Persistent,
         }
     }
 }
 
 #[derive(Debug, Clone, Default)]
 pub struct SysmonDecoder;
 
-impl PayloadDecoder<Vec<Event>> for SysmonDecoder {
+impl<'a> PayloadDecoder<Vec<SysmonEvent<'static>>> for SysmonDecoder {
     type DecoderError = SysmonDecoderError;
 
     #[tracing::instrument(skip(self), err)]
-    fn decode(&mut self, body: Vec<u8>) -> Result<Vec<Event>, Self::DecoderError> {
+    fn decode(&mut self, body: Vec<u8>) -> Result<Vec<SysmonEvent<'static>>, Self::DecoderError> {
         let decompressed = grapl_service::decoder::decompress::maybe_decompress(body.as_slice())?;
 
         let mut first_deserialization_error: Option<SysmonDecoderError> = None;
 
-        /*
-           This iterator is taking a set of bytes of the logs, splitting the logs on newlines,
-           converting the byte sequences to utf-8 strings, and then filtering on supported event
-           types: Process Creation, Network Connection, and File Creation.
+        use sysmon_parser::system::EventId;
 
-           https://docs.microsoft.com/en-us/sysinternals/downloads/sysmon#events
-        */
-        let events: Vec<_> = decompressed
-            .split(|byte| *byte == b'\n')
-            .map(String::from_utf8_lossy)
-            .map(|s| s.to_string())
-            .filter_map(|event_str| {
-                let parsed_event = Event::from_str(&event_str);
-                match parsed_event {
-                    Ok(event) => {
+        let xml = String::from_utf8_lossy(decompressed.as_slice());
```

Comment:
I think you can use str::from_utf8_lossy? but idk if that's better

Reply:
There is not a lossy variant for str, only String. FWIW `String::from_utf8_lossy` returns a Cow.

I'd rather not even use lossy, but I got a Utf8Error trying to use str::from_utf8. I thought I made event6.xml UTF8, so I was wondering if it was an artifact of the compression, which seems like it shouldn't be the case, but I also figured I shouldn't worry about changing that in this PR. The previous uses the same `String::from_utf8_lossy`.